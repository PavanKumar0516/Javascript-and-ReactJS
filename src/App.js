//import logo from './logo.svg';
import './App.css';

function App() {
  var o=
  {
        set a(value){
          this.val=value
        },
        get a(){
          return this.val;
        }
  }
  o.a=80;
 // console.log(" o.a ="+o.a);
  if(o.a==80){
    o.a=o.a+20;
  }
  return (
    <div className="App">  
    Welcome   App  Vaishali  
    <h2 data-testid="compId"> Janmashtami</h2>
    </div>
  );
}

export default App;
