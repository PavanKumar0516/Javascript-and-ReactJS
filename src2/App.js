import './App.css';
import React from "react";
//import ShowAllEmpComp from './Component/ShowAllEmpComp';
//import {​​​BrowserRouter as Router,Route,Link,NavLink}​​​ from "react-router-dom";
import UseEffectDemo from './Component/UseEffectDemo';
import CComponent from './Component/CComponenet';
//function App() {
  export const UserContext=React.createContext();
  export const UserChannel=React.createContext();
const App=()=>{
  return (<div>

   <UserContext.Provider value={'Pavankumar'}>
     <UserChannel.Provider value="Capgemini">
       <CComponent/>
     </UserChannel.Provider>
   </UserContext.Provider>
      
    </div>
  );}
//}

export default App;
