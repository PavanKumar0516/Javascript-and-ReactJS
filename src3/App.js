
import { BrowserRouter as Router,Switch,Route,Link,NavLink} from "react-router-dom";
import './App.css';
import Employee from './Component/Employee';
import React from 'react';
import ShowAllEmpComp from "./Component/ShowAllEmpComp";
import ShowAddEmpComp from "./Component/ShowAddEmpComp";
import ShowDeleteEmpComp from "./Component/ShowDeleteEmpComp";
import ShowUpdateEmpComp from "./Component/ShowUpdateEmpComp";
import 'bootstrap/dist/css/bootstrap.min.css';
function App() {
  
  //<li><Link to="/ShowDeleteEmpPage" class="text-white">Delete Emp</Link></li>
      //<Route path="/ShowDeleteEmpPage" component={ShowDeleteEmpComp}></Route>
  return (<div className="App"><Router>
      <div class="p-2 mb-2 bg-primary text-white"><h3>WELCOME TO EMPLOYEE MANAGEMENT!!!</h3>
      <li><Link to="/" class="text-white">Home</Link></li>
      <li><Link to="/ShowAddEmpPage" class="text-white">Add Emp</Link></li>
      <li><Link to="/" class="text-white">Show All Emp</Link></li>
      <li><Link to="/ShowUpdateEmpPage" class="text-white">Update Emp </Link></li>
      <li><Link to="/ShowDeleteEmpPage" class="text-white">Delete Emp </Link></li>
      </div>
      <hr/>
      <Route path="/" component={ShowAllEmpComp}></Route>
      <Route path="/ShowAllEmpPage" component={ShowAllEmpComp}></Route>
      <Route path="/ShowAddEmpPage" component={ShowAddEmpComp}></Route>
      <Route path="/ShowUpdateEmpPage/:id" component={ShowUpdateEmpComp}></Route>
      <Route path="/ShowDeleteEmpPage/:id" component={ShowDeleteEmpComp}></Route></Router>
      <div class="p-5 mb-2 bg-secondary"></div>
      </div>
    );
}

export default App;
