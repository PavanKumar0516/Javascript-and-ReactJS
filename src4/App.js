import logo from './logo.svg';
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import Register from './components/Register';
import Footer from './components/Footer';
import Header from './components/Header';
function App() {

  
  return (
    <div className="App">
      <header><Header/></header>
                 <body><Register/></body>
                 <footer ><Footer/></footer>
    </div>
  );
}

export default App;
