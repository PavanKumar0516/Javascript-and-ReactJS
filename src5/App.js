import logo from './logo.svg';
import './App.css';
import ViewEmployee from './component/ViewEmployee';
import AddEmployee from './component/AddEmployee';
import UpdateEmployee from './component/UpdateEmployee';
import DeleteEmployee from './component/DeleteEmployee';
import { BrowserRouter as Router,Switch,Route,Link,NavLink} from "react-router-dom";

function App() {
  return (
    <div className="App"><Router>
     <div class="p-2 mb-2 bg-primary text-white">
       <h3>WELCOME TO EMPLOYEE MANAGEMENT!!!</h3>
      <li><Link to="/" class="text-white">Home</Link></li>
      <li><Link to="/ShowAddEmpPage" class="text-white">Add Emp</Link></li>
      <li><Link to="/" class="text-white">Show All Emp</Link></li>
      <li><Link to="/ShowUpdateEmpPage" class="text-white">Update Emp </Link></li>
      <li><Link to="/ShowDeleteEmpPage" class="text-white">Delete Emp </Link></li>
      </div>
      <hr/>
      <Route path="/" component={ViewEmployee}></Route>
      <Route path="/ShowAllEmpPage" component={ViewEmployee}></Route>
      <Route path="/ShowAddEmpPage" component={AddEmployee}></Route>
      <Route path="/ShowUpdateEmpPage/:id" component={UpdateEmployee }></Route>
      <Route path="/ShowDeleteEmpPage/:id" component={DeleteEmployee}></Route></Router>
      <div class="p-5 mb-2 bg-secondary"></div></div>)
    }

export default App;
