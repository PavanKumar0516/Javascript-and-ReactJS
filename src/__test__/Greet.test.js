import React from "react";
import App from "../App";
import { render, screen,fireEvent } from '@testing-library/react';
import renderer from "react-test-renderer"

const tree =renderer.create(<App/>);
//-------------------------------
it("Test App componet renders correctly or not" ,
()=>{
    expect(tree).toMatchSnapshot();
});
//---------------------------------------------------
it(" Test app rendewrs h2 tag with id",()=>{
    const {getByTestId}=render(<App/>);
    expect(getByTestId("compId")).toHaveTextContent("Happy Janmashtami");
});