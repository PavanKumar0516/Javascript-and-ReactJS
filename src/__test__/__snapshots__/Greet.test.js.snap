// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App componet renders correctly or not 1`] = `
<div
  className="App"
>
  Welcome   App  Vaishali
  <h2
    data-testid="compId"
  >
    Happy Janmashtami
  </h2>
</div>
`;
